ai/agents.py

```diff
@@ -8,13 +9,17 @@
 
 
 class SimpleGuard(GuardAgent):
+    def __init__(self) -> None:
+        super().__init__()
+        self.type = 'SimpleGuard'
```

Comment:
Not a complaint but simply something that might be good to know. You can get the name of a class like this: `self.__class__.__name__`. It might be possible to do this once in the base class init so that you don't have to set the type manually in every sub class, but I'm not 100% sure (more like 85% :p) that it'd return the name of the subclass and not the parent class.

Reply:
that's indeed good to know, thanks! i'll test it some time